geozero/src/wkt/wkt_reader.rs

```diff
@@ -0,0 +1,344 @@
+use crate::error::{GeozeroError, Result};
+use crate::{FeatureProcessor, GeomProcessor, GeozeroDatasource, GeozeroGeometry};
+
+use std::io::Read;
+use wkt::types::{
+    Coord, GeometryCollection, LineString, MultiLineString, MultiPoint, MultiPolygon, Point,
+    Polygon,
+};
+use wkt::Geometry;
+
+/// WKT String.
+#[derive(Debug)]
+pub struct WktString(pub String);
+
+impl GeozeroGeometry for WktString {
+    fn process_geom<P: GeomProcessor>(&self, processor: &mut P) -> Result<()> {
+        read_wkt(&mut self.0.as_bytes(), processor)
+    }
+}
+
+/// WKT String slice.
+pub struct WktStr<'a>(pub &'a str);
+
+impl GeozeroGeometry for WktStr<'_> {
+    fn process_geom<P: GeomProcessor>(&self, processor: &mut P) -> Result<()> {
+        read_wkt(&mut self.0.as_bytes(), processor)
+    }
+}
+
+impl GeozeroDatasource for WktStr<'_> {
+    fn process<P: FeatureProcessor>(&mut self, processor: &mut P) -> Result<()> {
+        read_wkt(&mut self.0.as_bytes(), processor)
+    }
+}
+
+/// Wkt Reader.
+pub struct WktReader<'a, R: Read>(pub &'a mut R);
+
+impl<'a, R: Read> GeozeroDatasource for WktReader<'a, R> {
+    fn process<P: FeatureProcessor>(&mut self, processor: &mut P) -> Result<()> {
+        read_wkt(&mut self.0, processor)
+    }
+}
+
+/// Read and process WKT geometry.
+pub fn read_wkt<R: Read, P: GeomProcessor>(reader: &mut R, processor: &mut P) -> Result<()> {
+    use std::str::FromStr;
+    // PERF: it would be good to avoid copying data into this string when we already
+    // have a string as input. Maybe the wkt crate needs a from_reader implementation.
+    let mut wkt_string = String::new();
```

Comment:
Most of this file was copied from the geojson_reader. Like there we copy the input into a String, which seems wasteful, but I don't know of another way around it.

I don't think the wkt crate currently exposes anything other than a from_str constructor.

Does this seem reasonable for now?